File: my-solutions/0001-two-sum/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    seen = {}
    nums.each_with_index do |num, index|
        need_number = target - num
        if seen.key?(need_number)
        return [seen[need_number],index]
        else
        seen[num] = index
        end
    end
end

File: my-solutions/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int need_number = target - num;
            if (map.containsKey(need_number)) 
                return new int[] { map.get(need_number), i };
            else 
                map.put(nums[i], i);
        }
        return new int[] {};
    }
}
